prost/src/encoding.rs

```diff
@@ -519,6 +529,10 @@ fixed_width!(
     get_i64_le
 );
 
+const fn max_key_len_fits_in_size_of_elem<T>(_: &[T]) -> bool {
+    key_len(MAX_TAG) <= mem::size_of::<T>()
```

Comment:
This is a check to ensure that the encoding is "non-expanding", that is, it takes at least as much memory to allocate the data to be encoded as what it would take to encode it. So the length computation before which this is asserted cannot overflow `usize` and can be performed with unchecked arithmetic.

Reply:
I have added a comment explaining the purpose of the function in the updated commit 699bd9f9c854689d364364673dd65fd01a9c6266.